### PR TITLE
Change L2 redshift construct to L1

### PR DESCRIPTION
**Pull request by ejeffrli**
*Issue #, if available:*
L2 construct has issues with creating the proper parameter group names under the hood.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: validation_testing/cdk_federation_infra_provisioning/app/lib/stacks/redshift-stack.ts [hunk 1]
```diff
@@ -70,10 +79,34 @@ export class RedshiftStack extends cdk.Stack {
...
+        clusterSubnetGroupName: cfnClusterSubnetGroup.attrClusterSubnetGroupName,
+        publiclyAccessible: false,
+        vpcSecurityGroupIds: [securityGroup.securityGroupId],
+        clusterParameterGroupName: 'caseinsensitiveparametername'
+    });
 
     const s3Spill = new s3.Bucket(this, 'redshift_spill_location', {});
 
-    const connectionString = `jdbc:redshift://${cluster.clusterEndpoint.socketAddress}/test?user=athena&password=${password}`;
+    const connectionString = `jdbc:redshift://${cfnCluster.attrEndpointAddress}:${cfnCluster.attrEndpointPort}/test?user=athena&password=${password}`;
```
Comment (AbdulR3hman) [inline validation_testing/cdk_federation_infra_provisioning/app/lib/stacks/redshift-stack.ts:109]: example here.

File: validation_testing/cdk_federation_infra_provisioning/app/package.json [hunk 2]
```diff
@@ -13,15 +13,15 @@
...
     "@types/node": "18.11.18",
-    "aws-cdk": "2.63.2",
+    "aws-cdk": "2.118.0",
     "jest": "^29.4.1",
     "ts-jest": "^29.0.5",
     "ts-node": "^10.9.1"
   },
   "dependencies": {
     "@aws-cdk/aws-glue-alpha": "2.63.2-alpha.0",
     "@aws-cdk/aws-redshift-alpha": "2.63.2-alpha.0",
```
Comment (AbdulR3hman) [inline validation_testing/cdk_federation_infra_provisioning/app/package.json:23]: do we need this any more? given we don' use the L2 construct?

File: validation_testing/cdk_federation_infra_provisioning/app/package.json [hunk 3]
```diff
@@ -13,15 +13,15 @@
...
     "@types/jest": "^29.4.0",
     "@types/node": "18.11.18",
-    "aws-cdk": "2.63.2",
+    "aws-cdk": "2.118.0",
     "jest": "^29.4.1",
     "ts-jest": "^29.0.5",
     "ts-node": "^10.9.1"
   },
   "dependencies": {
     "@aws-cdk/aws-glue-alpha": "2.63.2-alpha.0",
```
Comment (AbdulR3hman) [inline validation_testing/cdk_federation_infra_provisioning/app/package.json:22]: can you also upgrade this version to keep it the same as the normal library; and test please.

**Review by AbdulR3hman — CHANGES_REQUESTED** [review]

File: validation_testing/cdk_federation_infra_provisioning/app/package.json (hunk 2 above)
Comment (ejeffrli) [reply validation_testing/cdk_federation_infra_provisioning/app/package.json:23]: yes we do need this. the 2.63.2 L1 construct doesn't define groupParameterName field

**Review by AbdulR3hman — APPROVED** [review]

**Pull request merged**